### ISSUE TITLE
Missing Test Case - 641. Design Circular Deque

### ISSUE THREAD

**Issue by Parshwa1234** [LeetCode-Feedback/LeetCode-Feedback#24451]
### LeetCode Username

ParshwaJain12

### Problem Number, Title, and Link

https://leetcode.com/problems/design-circular-deque/description/

### Bug Category

Missing test case *(Incorrect/Inefficient Code getting accepted because of missing test cases)*

### Bug Description

Basically if we create one size circular queue and insert one element and ask if its full or not then start and end point would be equal and it give wrong answer if is full function is incorrect for handling edge cases like my code:-

### Language Used for Code

C++

### Code used for Submit/Run operation

```bash
class MyCircularDeque {
public:
    vector<int>v;
    int start,end,sz;
    MyCircularDeque(int k) {
        v.resize(k,-1);
        start=-1,end=-1,sz=k;
    }
    
    bool insertFront(int value) {
        if(start==-1)
        {
            start=0;
            end=0;
            v[start]=value;
            return true;
        }
        else{
            if(isFull())
            {
                return 0;
            }
            else
            {
                --start;
                start+=sz;
                start%=sz;
                v[start]=value;
                return true;
            }

        }
    }
    
    bool insertLast(int value) {
        if(end==-1)
        {
            start=0,end=0;
            v[end]=value;
            return true;
        }
        else
        {
            if(isFull())
            {
                return 0;
            }
            else
            {
                end=end+1;
                end%=sz;
                v[end]=value;
                return 1;
            }
        }
        
    }
    
    bool deleteFront() {
        if(isEmpty())
        {
            return 0;
        }
        else
        {
           if(end==start)
           {
              start=-1;
              end=-1;
           }
           else
           {
            ++start;
            start%=sz;
           }
           return 1;
        }
    }
    
    bool deleteLast() {
        if(isEmpty())
        {
            return 0;
        }
        else
        {
            if(start==end)
            {
                start=-1;
                end=-1;
            }
            else
            {
                end=end-1;
                end+=sz;
                end%=sz;
            }
            return 1;
        }
    }
    
    int getFront() {
        if (isEmpty()) return -1;
        return v[start];
    }
    
    int getRear() {
        if (isEmpty()) return -1;
        return v[end];
    }
    
    bool isEmpty() {
        if(start==-1 && end==-1)
        {
            return 1;
        }
        return 0;
    }
    
    bool isFull() {
        if(start>end && end==start-1)
        {
            return 1;
        }
        else if(end>start && end-start+1==sz)
        {
            return 1;
        }
        else
        {
            return 0;
        }
    }
};

/**
 * Your MyCircularDeque object will be instantiated and called as such:
 * MyCircularDeque* obj = new MyCircularDeque(k);
 * bool param_1 = obj->insertFront(value);
 * bool param_2 = obj->insertLast(value);
 * bool param_3 = obj->deleteFront();
 * bool param_4 = obj->deleteLast();
 * int param_5 = obj->getFront();
 * int param_6 = obj->getRear();
 * bool param_7 = obj->isEmpty();
 * bool param_8 = obj->isFull();
 */
```

### Expected behavior

for test case :
Input
["MyCircularDeque","insertLast","isFull"]
[[1],[2],[]]
Output
[null,true,false]
Expected
[null,true,true]

### Screenshots

![image](https://github.com/user-attachments/assets/d584e07d-a864-41fc-963c-d8e2ddc088d3)

### Additional context

So basically this is an edge case where k=1 and start point =end point ... Also we can fix this code by adding proper check condition for isFull that is if start==((end+1)%k)
